packages/styled/types/base.d.ts

```diff
@@ -73,7 +73,10 @@ export interface CreateStyledComponent<
   <AdditionalProps extends {} = {}>(
     ...styles: Array<
       Interpolation<
-        ComponentProps & SpecificComponentProps & StyleProps & AdditionalProps
+        ComponentProps &
+          SpecificComponentProps &
+          StyleProps &
+          AdditionalProps & { theme: Theme }
```

Comment:
`{ theme: Theme }` gets always passed in as `StyleProps`, so we should be able to remove this from here, but when I try to do that then I get such errors:
```
Error: /emotion/packages/styled/types/tests-base.tsx:156:32
ERROR: 156:32  expect  TypeScript@next compile error: 
Object is possibly 'undefined'.

/emotion/packages/styled/types/tests.tsx:19:7
ERROR: 19:7    expect  Expected type to be:
  { theme?: Theme | undefined; } & ClassAttributes<HTMLDivElement> & HTMLAttributes<HTMLDivElement> & { bar: string; } & { theme: Theme; }
got:
  { theme?: Theme | undefined; } & ClassAttributes<HTMLDivElement> & HTMLAttributes<HTMLDivElement> & { bar: string; }
```

Still investigating this, but right now I don't see why this fails.

Reply:
While I don't understand why this fails - it might actually be a proper thing to specify this here instead of passing it from the above level in `StyleProps`. After all - `theme` is **always** available in interpolations, so it shouldn't be possible to "cancel" it out by passing StyleProps without it?

Reply:
Ok - I know why this thing has been failing:
https://github.com/joltmode/emotion/blob/464922628c9197988ac7ae38d33b84b6eb6d63d8/packages/styled/types/base.d.ts#L89-L93
This doesn't include `StyleProps`. It should right?

Regardless of that - my previous question stands. Should we pass Theme as part of StyleProps from the above or should we just pass it directly to Interpolation?

Reply:
Ok, I've decided to move `{ theme: Theme }` inline where parameters are being passed to Interpolation inside styled components. Seems the most appropriate solution - but I'm happy to discuss this further, maybe I've missed something? 

More than that - I'm right now kinda questioning if we need StyleProps at all, what do you think?